docs/handcraft.rst

```diff
@@ -0,0 +1,2049 @@
+      signature(s)
+    * **asset id**
+    * **metadata id**
+    * any optional value, such as ``version`` which defaults to ``1``
+
+    This makes the assembling of a payload more involved as one needs to
+    provide all values regardless of whether there are defaults or not.
+
+
+The transaction body
+--------------------
+The transaction body is made up of the following keys:
+
+* ``asset`` -- ``dict``
+* ``metadata`` -- ``dict``
+* ``operation`` -- ``str``
+* ``conditions`` -- ``list`` of ``dict``
+* ``fulfillments`` -- ``list`` of ``dict``
+
+asset
+^^^^^
+
+.. ipython::
+
+    asset = {
+        'data': {},
+        'divisible': False,
+        'refillable': False,
+        'updatable': False,
+        'id': '',
+    }
+
+.. note:: In a signature-less system, the values for the keys:
+    
+    * ``'divisible'``
+    * ``'refillable'``
+    * ``'updatable'``
+    * ``'id'``
+
+    could all be generated on the server side.
+
+Example of an asset payload:
+
+.. ipython::
+
+    asset = {
+        'data': {
+            'bicycle': {
+                'manufacturer': 'bkfab',
+                'serial_number': 'abcd1234',
+            },
+        },
+        'divisible': False,
+        'refillable': False,
+        'updatable': False,
+        'id': '7ab63c48-4c24-41df-a1bd-934bb609a7f7',
+    }
+
+
+metadata
+^^^^^^^^
+.. ipython::
+
+    metadata = {
+        'data': {},
+        'id': '',
+    }
+
+.. note:: In a signature-less system, the values for the keys:
+    
+    * ``'id'``
+
+    could be generated on the server side.
+
+Example of a metadata payload:
+
+.. ipython::
+
+    metadata = {
+        'data': {
+            'planet': 'earth',
+        },
+        'id': 'ad8c83bd-9192-43b3-b636-af93a3a6b07c',
+    }
+
+
+operation
+^^^^^^^^^
+
+.. ipython::
+
+    operation = '<operation>'
+
+``<operation>`` is one of ``'CREATE'``, ``'TRANSFER'``, or ``'GENESIS'``
+
+.. important::
+
+    Case sensitive, all letters must be capitalized.
+
+conditions
+^^^^^^^^^^
+The purpose of the condition is to lock the transaction, such that a valid
+fulfillment is required to unlock it. In the case of signature-based schemes,
+the lock is basically a public key, such that in order to unlock the
+transaction one needs to have the private key.
+
+Example of a condition payload:
+
+.. ipython::
+
+    {
+        'amount': 1,
+        'cid': 0,
+        'condition': {
+            'details': {
+                'bitmask': 32,
+                'public_key': '8L6ngTZ5ixuFEr1GiunrFNWtGkft4swWWArXjWJu2Uwc',
+                'signature': None,
+                'type': 'fulfillment',
+                'type_id': 4,
+            },
+            'uri': 'cc:4:20:bOZjTedaOgPsbYjh3QeOEQCj1o1lIvVefR71sS8egnM:96'
+        },
+        'owners_after': ['8L6ngTZ5ixuFEr1GiunrFNWtGkft4swWWArXjWJu2Uwc'],
+    }
+    
+
+fulfillments
+^^^^^^^^^^^^
+At the moment, we have to distinguish between fulfillment payloads **before**
```

Comment:
I find this paragraph a bit confusing. I understand what you are trying to say.
What actually want is to construct the fulfillment object but without the signature.

One reason to do this is to avoid exposure of the private key. I may want to use some utility to generate the payloads (e.g. `bigchaindb_driver`, `bdb-cli`) and then I want to sign the paylod if I feel everything is ok. Possibly on another machine where I store my private keys